cdap-app-templates/cdap-etl/cdap-etl-batch/src/main/java/co/cask/cdap/etl/batch/condition/BasicConditionContext.java

```diff
@@ -18,27 +18,95 @@
 
 import co.cask.cdap.api.TxRunnable;
 import co.cask.cdap.api.security.store.SecureStoreData;
+import co.cask.cdap.api.workflow.NodeValue;
 import co.cask.cdap.api.workflow.WorkflowContext;
+import co.cask.cdap.api.workflow.WorkflowToken;
 import co.cask.cdap.etl.api.condition.ConditionContext;
 import co.cask.cdap.etl.api.condition.StageStatistics;
 import co.cask.cdap.etl.common.AbstractStageContext;
+import co.cask.cdap.etl.common.Constants;
 import co.cask.cdap.etl.common.PipelineRuntime;
 import co.cask.cdap.etl.spec.StageSpec;
+import com.google.common.collect.ImmutableMap;
 import org.apache.tephra.TransactionFailureException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
 
 /**
  * Default implementation of the {@link ConditionContext}.
  */
 public class BasicConditionContext extends AbstractStageContext implements ConditionContext {
-
+  private static final Logger LOG = LoggerFactory.getLogger(BasicConditionContext.class);
   private final WorkflowContext context;
+  private final Map<String, StageStatistics> stageStatistics;
 
   public BasicConditionContext(WorkflowContext context, PipelineRuntime pipelineRuntime, StageSpec stageSpec) {
     super(pipelineRuntime, stageSpec);
     this.context = context;
+    this.stageStatistics = createStageStatistics(context);
+  }
+
+
+  private Map<String, StageStatistics> createStageStatistics(WorkflowContext context) {
+    Map<String, StageStatistics> stageStatistics = new HashMap<>();
+    WorkflowToken token = context.getToken();
+    for (WorkflowToken.Scope scope : Arrays.asList(WorkflowToken.Scope.SYSTEM, WorkflowToken.Scope.USER)) {
```

Comment:
In MR stage statistics are added as counters, which are added in `SYSTEM` scope, however spark accumulators are added in the USER scope. so we have to iterate through both